src/core/ddsi/src/ddsi_entity.c

```diff
@@ -40,6 +40,11 @@ int ddsi_compare_guid (const void *va, const void *vb)
   return memcmp (va, vb, sizeof (ddsi_guid_t));
 }
 
+int ddsi_compare_entityid (const void *va, const void *vb)
+{
+  return memcmp (va, vb, sizeof (ddsi_entityid_t));
```

Comment:
I'm curious what the performance difference between a `memcmp` and `return (a->u==b->u)?0:(a->u<b->u)?-1:1` would be ... https://godbolt.org/z/rfe6aahb1 suggests on 64-bit ARM with clang the difference will be really small, but on x64 with gcc (and who cares about x64 anyway? 😁), it is a call to `memcmp` vs a few handful instructions.

The difference will likely be meaningless compared to all the other things that happen, but if you don't mind making this change, then I would like it ever so slightly better, because I do prefer the other way.

Reply:
Fixed in [45353d8](https://github.com/eclipse-cyclonedds/cyclonedds/pull/1378/commits/45353d8bdbd47230f17555794c945c3678780302)